File: frontend/src/app/services/review.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Review, ReviewRequest } from '../models/review.model';

@Injectable({ providedIn: 'root' })
export class ReviewService {
    private http = inject(HttpClient);
    private readonly baseUrl = `${environment.apiUrl}/conference-service/api/v1`;

    getReviewsByConference(conferenceId: string): Observable<Review[]> {
        return this.http.get<Review[]>(`${this.baseUrl}/conferences/${conferenceId}/reviews`);
    }

    getReviewById(reviewId: string): Observable<Review> {
        return this.http.get<Review>(`${this.baseUrl}/reviews/${reviewId}`);
    }

    createReview(conferenceId: string, review: ReviewRequest): Observable<Review> {
        return this.http.post<Review>(`${this.baseUrl}/conferences/${conferenceId}/reviews`, review);
    }

    updateReview(reviewId: string, review: ReviewRequest): Observable<Review> {
        return this.http.put<Review>(`${this.baseUrl}/reviews/${reviewId}`, review);
    }

    deleteReview(reviewId: string): Observable<void> {
        return this.http.delete<void>(`${this.baseUrl}/reviews/${reviewId}`);
    }
}

File: backend/conference-service/src/main/java/ma/enset/conferenceservice/service/ReviewService.java
package ma.enset.conferenceservice.service;

import ma.enset.conferenceservice.dto.ReviewRequestDto;
import ma.enset.conferenceservice.dto.ReviewResponseDto;

import java.util.List;
import java.util.UUID;

public interface ReviewService {
    ReviewResponseDto createReview(UUID conferenceId, ReviewRequestDto reviewRequestDto);
    ReviewResponseDto getReviewById(UUID reviewId);
    List<ReviewResponseDto> getReviewsByConferenceId(UUID conferenceId);
    ReviewResponseDto updateReview(UUID reviewId, ReviewRequestDto reviewRequestDto);
    void deleteReviewById(UUID reviewId);
}
